build.sbt

```diff
@@ -1575,7 +1582,7 @@ lazy val `runtime-with-polyglot` =
     .configs(Benchmark)
     .settings(
       frgaalJavaCompilerSetting,
-      inConfig(Compile)(truffleRunOptionsSettings),
+      inConfig(Compile)(truffleRunOptionsNoAssertSettings),
```

Comment:
There is a problem with _Invalid Sharing of AST nodes_ in `EpbContext` - it will need more time to solve it. Disabling the asserts for now in the `runtime-with-polyglot` project.

```
"job-pool-1"
	at com.oracle.truffle.polyglot.PolyglotSharingLayer.invalidSharingError(PolyglotSharingLayer.java:649)
	at com.oracle.truffle.polyglot.PolyglotFastThreadLocals.validSharing(PolyglotFastThreadLocals.java:266)
	at com.oracle.truffle.polyglot.PolyglotFastThreadLocals.getLanguageContext(PolyglotFastThreadLocals.java:236)
	at com.oracle.truffle.polyglot.PolyglotFastThreadLocals$ContextReferenceImpl.get(PolyglotFastThreadLocals.java:457)
	at org.enso.interpreter.epb.EpbContext.get(EpbContext.java:57)
	at org.enso.interpreter.epb.node.ForeignEvalNode.getEpbContext(ForeignEvalNode.java:186)
	at org.enso.interpreter.epb.node.ForeignEvalNode.parseJs(ForeignEvalNode.java:118)
	at org.enso.interpreter.epb.node.ForeignEvalNode.lockAndParse(ForeignEvalNode.java:83)
	at org.enso.interpreter.epb.node.ForeignEvalNode.ensureParsed(ForeignEvalNode.java:64)
	at org.enso.interpreter.epb.node.ForeignEvalNode.execute(ForeignEvalNode.java:53)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:709)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:632)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:565)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:549)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:495)
	at org.graalvm.compiler.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:68)
	at org.enso.interpreter.node.expression.foreign.ForeignMethodCallNode.executeGeneric(ForeignMethodCallNode.java:51)
	at org.enso.interpreter.node.callable.function.StatementNode.executeGeneric(StatementNode.java:53)
	at org.enso.interpreter.node.callable.function.BlockNode.executeGeneric(BlockNode.java:56)
	at org.enso.interpreter.node.ExpressionNodeWrapper.executeGeneric(ExpressionNodeWrapper.java:114)
	at org.enso.interpreter.node.ClosureRootNode.execute(ClosureRootNode.java:85)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:709)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:632)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:565)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:549)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callDirect(OptimizedCallTarget.java:495)
	at org.graalvm.compiler.truffle.runtime.OptimizedDirectCallNode.call(OptimizedDirectCallNode.java:68)
	at org.enso.interpreter.node.callable.ExecuteCallNode.callDirect(ExecuteCallNode.java:56)
	at org.enso.interpreter.node.callable.ExecuteCallNodeGen.executeAndSpecialize(ExecuteCallNodeGen.java:90)
	at org.enso.interpreter.node.callable.ExecuteCallNodeGen.executeCall(ExecuteCallNodeGen.java:52)
	at org.enso.interpreter.node.callable.dispatch.CurryNode.doCall(CurryNode.java:155)
	at org.enso.interpreter.node.callable.dispatch.CurryNode.execute(CurryNode.java:108)
	at org.enso.interpreter.node.callable.dispatch.InvokeFunctionNode.invokeCached(InvokeFunctionNode.java:92)
	at org.enso.interpreter.node.callable.dispatch.InvokeFunctionNodeGen.executeAndSpecialize(InvokeFunctionNodeGen.java:102)
	at org.enso.interpreter.node.callable.dispatch.InvokeFunctionNodeGen.execute(InvokeFunctionNodeGen.java:58)
	at org.enso.interpreter.node.callable.InvokeMethodNode.doFunctionalDispatch(InvokeMethodNode.java:107)
	at org.enso.interpreter.node.callable.InvokeMethodNodeGen.executeAndSpecialize(InvokeMethodNodeGen.java:201)
	at org.enso.interpreter.node.callable.InvokeMethodNodeGen.execute(InvokeMethodNodeGen.java:181)
	at org.enso.interpreter.node.callable.InvokeCallableNode.invokeDynamicSymbol(InvokeCallableNode.java:247)
	at org.enso.interpreter.node.callable.InvokeCallableNodeGen.executeAndSpecialize(InvokeCallableNodeGen.java:98)
	at org.enso.interpreter.node.callable.InvokeCallableNodeGen.execute(InvokeCallableNodeGen.java:63)
	at org.enso.interpreter.node.callable.ApplicationNode.executeGeneric(ApplicationNode.java:97)
	at org.enso.interpreter.node.ClosureRootNode.execute(ClosureRootNode.java:85)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:709)


...
	at org.enso.interpreter.node.callable.InteropApplicationNodeGen$Uncached.execute(InteropApplicationNodeGen.java:168)
	at org.enso.interpreter.node.callable.FunctionCallInstrumentationNode$FunctionCall$Execute.callCached(FunctionCallInstrumentationNode.java:103)
	at org.enso.interpreter.node.callable.FunctionCallGen$InteropLibraryExports$Uncached.execute(FunctionCallGen.java:147)
	at com.oracle.truffle.api.interop.InteropLibrary$Asserts.execute(InteropLibrary.java:3034)
	at com.oracle.truffle.api.interop.InteropLibraryGen$UncachedDispatch.execute(InteropLibraryGen.java:6866)
	at org.enso.interpreter.service.ExecutionService.execute(ExecutionService.java:176)
	at org.enso.interpreter.service.ExecutionService.execute(ExecutionService.java:217)
	at org.enso.interpreter.instrument.job.ProgramExecutionSupport$.executeProgram(ProgramExecutionSupport.scala:133)
	at org.enso.interpreter.instrument.job.ProgramExecutionSupport$.$anonfun$runProgram$3(ProgramExecutionSupport.scala:243)
	at org.enso.interpreter.instrument.job.ProgramExecutionSupport$$$Lambda$1854.1336163654.apply$mcV$sp
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at cats.syntax.EitherObjectOps$.catchNonFatal$extension(either.scala:391)
	at org.enso.interpreter.instrument.job.ProgramExecutionSupport$.$anonfun$runProgram$2(ProgramExecutionSupport.scala:243)
	at org.enso.interpreter.instrument.job.ProgramExecutionSupport$$$Lambda$1853.1643507906.apply
	at scala.util.Either.flatMap(Either.scala:352)
	at org.enso.interpreter.instrument.job.ProgramExecutionSupport$.runProgram(ProgramExecutionSupport.scala:237)
	at org.enso.interpreter.instrument.job.ExecuteJob.run(ExecuteJob.scala:33)
	at org.enso.interpreter.instrument.job.ExecuteJob.run(ExecuteJob.scala:14)
	at org.enso.interpreter.instrument.execution.JobExecutionEngine.$anonfun$runInternal$1(JobExecutionEngine.scala:117)
	at org.enso.interpreter.instrument.execution.JobExecutionEngine$$Lambda$994.1595560891.call
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.truffle.polyglot.PolyglotThread.access$001(PolyglotThread.java:53)
	at com.oracle.truffle.polyglot.PolyglotThread$1.execute(PolyglotThread.java:100)
	at com.oracle.truffle.polyglot.PolyglotThread$ThreadSpawnRootNode.executeImpl(PolyglotThread.java:134)
	at com.oracle.truffle.polyglot.PolyglotThread$ThreadSpawnRootNode.execute(PolyglotThread.java:125)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.executeRootNode(OptimizedCallTarget.java:709)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.profiledPERoot(OptimizedCallTarget.java:632)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:565)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.doInvoke(OptimizedCallTarget.java:549)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callIndirect(OptimizedCallTarget.java:477)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.call(OptimizedCallTarget.java:458)
	at com.oracle.truffle.polyglot.PolyglotThread.run(PolyglotThread.java:96)
```

Reply:
Reported as [pivotal 184321951](https://www.pivotaltracker.com/story/show/184321951)